keras/regularizers.py

```diff
@@ -15,6 +16,28 @@ def __call__(self, loss):
     def get_config(self):
         return {'name': self.__class__.__name__}
 
+class EigenvalueRegularizer(Regularizer):
+    def __init__(self, k):
+        self.k = k
+        self.uses_learning_phase = True
+
+    def set_param(self, p):
+        self.p = p
+
+    def __call__(self, loss):
+        W = self.p
+        WW = T.dot(W.T,W)
+        dim1, dim2 = WW.shape.eval() # The number of neurons in the layer
+        k = self.k
+        o = np.ones(dim1) # initial values for the dominant eigenvector
+
+        # POWER METHOD FOR APPROXIMATING THE DOMINANT EIGENVECTOR (9 ITERATIONS):
+        domineigvec = T.dot(WW,T.dot(WW,T.dot(WW,T.dot(WW,T.dot(WW,T.dot(WW,T.dot(WW,T.dot(WW,T.dot(WW,o)))))))))
```

Comment:
The spaces were introduced and the equation of the dominant eigenvector was replaced by a for loop, where it is possible to define the number of iterations of the power method (the GPU/CPU time was evaluated and there is no difference due to this changing).